### ISSUE TITLE
incompatible override false positive on method that uses descriptor with contravariant generic

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#17736]
```py
# mypy: enable-incomplete-feature=NewGenericSyntax

from typing import Callable, override


class Foo[T]:
    def __init__(self, value: Callable[[T], None]) -> None:
        pass

    def __get__(self, instance: T, owner: type[T]) -> int:
        return 1


class Bar:
    @Foo
    def foo(self) -> None:
        pass


class Baz(Bar):
    @Foo
    @override
    def foo(self) -> None: # error (override)
        pass
```
```
main.py:23: error: Signature of "foo" incompatible with supertype "Bar"  [override]
main.py:23: note:      Superclass:
main.py:23: note:          Foo[Bar]
main.py:23: note:      Subclass:
main.py:23: note:          Foo[Baz]
```

although this error is usually correct because the generic is contravariant, in this case i believe it's safe because the descriptor prevents the `Foo` class from being accessible on instances of `Bar` and `Baz`:

```py
Bar().foo # int
Baz().foo # int
```

fyi [pyright does not report an error](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgDRRgBu1IISAJtQFA8DG5AZwFQAYmDABtACoBdAFxcoSqJ2BQA%2ButRIYmgBQDqJYPUakArtTmFS5KtQnSZ9AHJgU1GQEooAWgB8UG4eCsphUAhEQjxhqhrqaNS66gZGJpgoAjBEKLxWUFL0YADuHiDW8AgOsj4BGTCh4UogSeYgKFAAjDxc-FHCAEJE5YrKAAJiYKNKccDiqca1gcFW0%2BGR0XyCg0QAXnpDIF6NShPia2NMLGyca7Pzhot%2By%2B6rTUobAkA) in this case.
